src/ConnectedMode/Migration/MSBuildFileCleaner.cs

```diff
@@ -34,18 +37,68 @@ internal class MSBuildFileCleaner : IFileCleaner
         /// </summary>
         public const string Unchanged = null;
 
+        private const string AdditionalFilesTagName = "AdditionalFiles";
+
         private readonly ILogger logger;
+        private readonly IXmlDocumentHelper xmlDocumentHelper;
 
         [ImportingConstructor]
-        public MSBuildFileCleaner(ILogger logger)
+        public MSBuildFileCleaner(ILogger logger) : this(logger, new XmlDocumentHelper())
+        {
+        }
+
+        internal /* for testing */ MSBuildFileCleaner(ILogger logger, IXmlDocumentHelper xmlDocumentHelper)
         {
             this.logger = logger;
+            this.xmlDocumentHelper = xmlDocumentHelper;
         }
 
-        public Task<string> CleanAsync(string content, LegacySettings legacySettings, CancellationToken token)
+        public string Clean(string content, LegacySettings legacySettings, CancellationToken token)
         {
-            // TODO
-            return Task.FromResult(Unchanged);
+            var document = xmlDocumentHelper.LoadFromString(content);
+
+            var nodesToRemove = new List<XmlNode>();
+            foreach (XmlNode item in document.GetElementsByTagName(AdditionalFilesTagName))
+            {
+                if (ContainsSonarLintXmlReferenceInAttributes(item.Attributes,
+                        legacySettings.PartialVBSonarLintXmlPath,
+                        legacySettings.PartialCSharpSonarLintXmlPath))
+                {
+                    LogVerbose("Detected SonarLint.xml: " + item.Value);
+                    nodesToRemove.Add(item);
+                }
+            }
+
+            if (!nodesToRemove.Any())
+            {
+                logger.LogVerbose("No settings to remove");
+                return Unchanged;
+            }
+
+            logger.WriteLine(MigrationStrings.Cleaner_RemovingSettings, nodesToRemove.Count);
+
+            nodesToRemove.ForEach(node =>
+            {
+                Debug.Assert(node.ParentNode != null);
+                node.ParentNode.RemoveChild(node);
+            });
+
+            return xmlDocumentHelper.SaveToString(document);
+        }
+
+        private static bool ContainsSonarLintXmlReferenceInAttributes(XmlAttributeCollection attributeCollection,
+            params string[] sonarlintXmlPaths)
+        {
+            // Matches the following:
+            // 		<AdditionalFiles Include="..\..\.sonarlint\my_project_key\CSharp\SonarLint.xml" />
+            return attributeCollection
+                .Cast<XmlAttribute>()
+                .Any(attribute =>
+                    sonarlintXmlPaths.Any(path =>
+                        attribute.Name == "Include" &&
```

Comment:
I added the condition about the `attribute.Name`, so we will match

  `<AdditionalFiles Include="..." ... />`

but not 

  `<AdditionalFiles Remove="..." ... />`